### PR TITLE
Update i/o routines for new surface coldstart file.

### LINKED ISSUE #670: Update fv3atm_sfc_io.F90 to read new surface coldstart files
## Description
The introduction of fractional grids required new surface-specific variables for roughness length, snow depth/liquid equivalent, column temperature, and skin temperature. For example, roughness length is stored in three separate variables: zorll (land), zorlw (open water) and zorli (ice). There is also a 'composite' or 'merged' roughness record, zorl. 

The i/o module (fv3atm_sfc_io.F90) will read in a composite field (mandatory) from the coldstart file and optionally the surface-specific fields.  If the surface-specific fields are not in the file, the module will populate them from the composite value. This allows for cold starting with files from the current version of chgres_cube. 

## Solution
The chgres_cube program has been updated to recognize fractional grids. As part of this update, it will always output the surface-specific records. Update the model i/o module to read them, then, compute the composite field (some composite fields are used by the physics).

## Testing:
So far, tested on Hera with the following configurations: 
- C96 and C384 resolutions
- Noah and NoahMP land model.
- Fractional and non-fraction grids.

## Dependent issues:
https://github.com/ufs-community/UFS_UTILS/issues/123

### PR DESCRIPTION
**Pull request by GeorgeGayno-NOAA**
## Description
The UFS_UTILS chgres_cube program is being updated to process fractional grids. The surface coldstart file generated by the new chgres_cube will contain new records. Update the FV3 routines to read either the new surface coldstart format or the original format. The new coldstart files (named version 2) will contain the following global attribute:
 
```
// global attributes:
                :file_version = "V2" ;
```

The model i/o routine will check for this attribute and act accordingly.

Since the updates to the model i/o routine are backwards compatible, regression test results should not change.

### Issue(s) addressed
- Fixes #670.

### Associated PR's
- ufs-community/UFS_UTILS#873
- ufs-community/ufs-weather-model#2005

## Testing
The regression tests were run on Hera using the branch at bbc5bf8 and passed. The regression test log is here: https://github.com/GeorgeGayno-NOAA/ufs-weather-model/blob/9bfffc16568b5d2039c956642d601b41dbad9190/tests/logs/RegressionTests_hera.log

## Dependencies
The updates to chgres_cube are being tracked here: https://github.com/ufs-community/UFS_UTILS/issues/123

This PR will also include one or two new regression tests. See below for details.

### CODE REVIEW

**Comment by GeorgeGayno-NOAA** [comment]
Request one new regression test that is based on the 'control_c48_intel' test. That test is ATM only and uses a non-fractional grid. 

The new test's coldstart files used the same input GFS data to chgres (located here `/scratch1/NCEPDEV/stmp2/Jun.Wang/gdas/2021032206`) and the same model configuration files. 

The 'oro' and other fixed files used by the new test were: `/scratch1/NCEPDEV/global/glopara/fix/orog/20220805/C48.mx500`.     

The coldstart files for the new test are here:  `/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/chgres_cube.fractional/RT/FV3_input_data48/INPUT_L127.v2.sfc`

**Comment by GeorgeGayno-NOAA** [comment]
> Request one new regression test that is based on the 'control_c48_intel' test. That test is ATM only and uses a non-fractional grid.
> 
> The new test's coldstart files used the same input GFS data to chgres (located here `/scratch1/NCEPDEV/stmp2/Jun.Wang/gdas/2021032206`) and the same model configuration files.
> 
> The 'oro' and other fixed files used by the new test were: `/scratch1/NCEPDEV/global/glopara/fix/orog/20220805/C48.mx500`.
> 
> The coldstart files for the new test are here: `/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/chgres_cube.fractional/RT/FV3_input_data48/INPUT_L127.v2.sfc`

For details on the new regression test data, see https://github.com/ufs-community/ufs-weather-model/pull/2005

**Comment by GeorgeGayno-NOAA** [comment]
@jkbk2004 - This is related to https://github.com/ufs-community/ufs-weather-model/pull/2005. I do not have permissions to assign a reviewer to this PR.

**Review by DusanJovic-NOAA — APPROVED** [review]

File: io/fv3atm_restart_io.F90 [hunk 1]
```diff
@@ -644,8 +646,21 @@ subroutine sfc_prop_restart_read (Sfcprop, Atm_block, Model, fv_domain, warm_sta
...
     if( .not.amiopen ) call mpp_error(FATAL, 'Error opening file'//trim(infile))
 
+    if (global_att_exists(Sfc_restart, "file_version")) then
+      call get_global_attribute(Sfc_restart, "file_version", file_ver)
+      if (file_ver == "V2") then
+        sfc%is_v2_file=.true.
+      endif
+    endif
+
+    print*,'^^^^ process sfc file ',sfc%is_v2_file
```
Comment (GeorgeGayno-NOAA) [inline io/fv3atm_restart_io.F90]: Looks like I left some diagnostic prints. Should I remove all of them?

Comment (BrianCurtis-NOAA) [reply io/fv3atm_restart_io.F90]: That would be appreciated. Thanks.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Comment by zach1221** [comment]
@jkbk2004 we have approvals here, and testing is complete on 2005. Can you please merge this PR?

**Pull request merged**